### PR TITLE
Book-appointment_back-end: Dev

### PR DESCRIPTION
**Pull request by jahongiry**
## In this PR, we have done the following requirements:
- The user logs in to the website, only by typing the username
- In the navigation panel, the user can see links to:
"Reserve" form.
"My reservations".
"Add cars that you selected as a theme" (the link is visible to everybody).
- On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).
- When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).
- On the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).
- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.
**Team: @nottherealalanturing @myelin0 @jahongiry @Zelalem1222**
Front end pull request link: https://github.com/jahongiry/book-appointment_front-end/pull/10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,120 @@
...
+
+* Run the project in your browser:
+
+```
+rails server
+```
+
+## Run Tests
+
+- To run tests, Run the following command on your terminal
```
Comment (Meltrust) [inline README.md:53]: - [ ] As per your instructions, you are using Rspec, but the "test" folder still exists.  Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @jahongiry,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  API is working ✔️ 
-  Good readme ✔️ 

You can do it

## Suggested Changes ♻️
_Check the comments under the review._

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: README.md [hunk 2]
```diff
@@ -1 +1,120 @@
-# book-appointment_back-end
\ No newline at end of file
+# BOOK_APPOINTMENT-BACK-END
```
[the remark this answers is not in the document]
Comment (myelin0) [reply README.md:1]: Thank you for this suggestion. We added the swagger docs screenshot.:ok_hand:

File: README.md [hunk 3]
```diff
@@ -1 +1,120 @@
...
+
+## Deployment
+
+The server was deployed on heroku, and runs on [this domain](_).
+The above endpoint permits you to see all cars available for reservation. Checkout the api documentation below for the other endpoints.
+
+## API Documentation
+
+Visit [this endpoint](http://localhost:3000/api-docs/index.html) to see a complete documentation of all available endpoints for this API.
+We hope this helps you find your way around our API.
```
[the remark this answers is not in the document]
Comment (myelin0) [reply README.md:69]: Thanks! Documentation is now added.:+1:

**Review by Meltrust — CHANGES_REQUESTED** [review]
## Kindly go to the [frontend PR](https://github.com/jahongiry/book-appointment_front-end/pull/10) to view the review results and the review comments.

Hi @jahongiry ,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  App is working well✔️ 
-  Readme has been corrected ✔️
-  All the backend review requests have been implemented ✔️ 
-  Documentation has been added to the API ✔️ 
-  Nice work ✔️  

You can do it

## Suggested Changes ♻️


You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @jahongiry,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


   - Tests are passing✔️
   - App is working well✔️
   - Readme has been corrected ✔️
   - All the backend review requests have been implemented ✔️
   - Documentation has been added to the API ✔️
   - Superb job getting this far, congratulations 💪  ✔️✔️✔️✔️✔️✔️✔️


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Pull request merged**